apps/discovery_api/lib/discovery_api/event_handler.ex

```diff
@@ -46,12 +47,15 @@ defmodule DiscoveryApi.EventHandler do
     with {:ok, organization} <- DiscoveryApi.Schemas.Organizations.get_organization(dataset.technical.orgId),
          {:ok, _cached} <- SystemNameCache.put(dataset.id, organization.name, dataset.technical.dataName),
          model <- Mapper.to_data_model(dataset, organization) do
+
       DiscoveryApi.Search.Storage.index(model)
       save_dataset_to_recommendation_engine(dataset)
       ResponseCache.invalidate()
       Logger.debug(fn -> "Successfully handled message: `#{dataset.technical.systemName}`" end)
-
       merge(:models, model.id, model)
+      DataJson.delete_data_json()
```

Comment:
Why do this after the update to view state?  Shouldn't the merge be last?  (Tony)

Reply:
This is to deal with a possible race condition where:
- data.json is deleted
- someone requests it via API and it gets rebuilt (without the new/updated model from the merge)
- new/updated model is merged, but not included in the now cached data.json file

Merging first ensures that the model is available for any rebuilds of the data.json file that might occur after the delete.

- Ben and Kevin